Fix docs as says bug 9085

Yajo opened the pull request:
This fixes #9085 (well... documents it).

Yajo commented:
cc @sle-odoo

sle-odoo commented:
i think you don't get it entirely. It's not a "false doc" (as your commit message tells) nor it is a "lie" (as you wrote in #9085). You deleted a sample code explaining how to add css rules available in the reports by documenting how to troubleshoot a common issue in an odoo setup. While i agree the information you add are not superfluous, you have to keep the first part (and maybe put it in a danger block && finish it with how to add a link to a stylesheet).

Also, i'm not okay with the wording ("awesomely ugly", why not simply "if it appears that your report is missing the styles (i.e. the text appears but the style/layout is different from the html version)"). You could also write that, to know if it is actually this issue, it is useful to check the server logs and see that the server does not receive request for the report's assets.

There is also a typo, the last "web.base.url" should be "report.url". And maybe you could document that web.base.url is reset everytime the admin is logged (to check) and whate does the web.base.url.freeze key do too? So it i's complete.

Regards

sle-odoo commented:
i mean put the troubleshoot part in a danger block. and now that i think about it, it may not be the right place for this text (troubleshoot part).. i guess it should land in the deploy guide, next to the version of wkhtmltopdf supported.

Yajo commented:
I agree to all, except that I deleted instructions that, if followed, did not work. I have not found any alternative way to do it, so I deleted it to avoid confusions.

I'll fix all other things, but if you want me to restore that paragraph, please provide a workaround or fix.

Yajo commented on doc/howtos/backend.rst:
```diff
@@ -1622,6 +1622,28 @@ http://localhost:8069/report/html/account.report_invoice/1 (if ``account`` is
 installed) and the PDF version through
 http://localhost:8069/report/pdf/account.report_invoice/1.
 
+.. _reference/backend/reporting/printed-reports/pdf-without-styles:
```
All done now. Is this OK? I never wrote such things with Sphinx...

sle-odoo commented:
yes you are right, actually i we have to make an xpath expr="." position="inside" inherit_id="report.style" to directly add css rules, but this doesn't work with lxml/our xpath things so we have to put the css rules in a t tag

```
<t>
.rule1{color:red;}
</t>
```

this should work. Could you try it please?

Yajo commented:
That works, let me update this.

rafaelbn commented:
Hi, how is the state of this PR? It looks :+1:

Yajo commented:
I don't want to bother anyone, but this is ready to merge IMHO.

sle-odoo commented:
merged at 4eab4d786805ecfd9bdff0b013ff831c4984cc99

thanks a lot!
